src/Compilers/Core/Portable/MetadataReference/AssemblyIdentity.DisplayName.cs

```diff
@@ -162,6 +162,18 @@ public static bool TryParseDisplayName(string displayName, [NotNullWhen(true)] o
             return TryParseDisplayName(displayName, out identity, parts: out _);
         }
 
+        /// <summary>
+        /// Perf: ETW traces show 2%+ of all allocations parsing assembly identity names.  This is due to how large
+        /// these strings can be (600+ characters in some cases), and how many substrings are continually produced as
+        /// the string is broken up into the pieces needed by AssemblyIdentity.  The capacity of 1024 was picked as
+        /// around 700 unique strings were found in a solution the size of Roslyn.sln.  So this seems like a reasonable
+        /// starting point for a large solution.  This cache takes up around 240k in memory, but ends up saving >80MB of
+        /// garbage over typing even a few characters.  And, of course, that savings just grows over the lifetime of a
+        /// session this is hosted within.
+        /// </summary>
+        private static readonly ConcurrentCache<string, (AssemblyIdentity? identity, AssemblyIdentityParts parts)> s_TryParseDisplayNameCache =
+            new ConcurrentCache<string, (AssemblyIdentity? identity, AssemblyIdentityParts parts)>(1024, ReferenceEqualityComparer.Instance);
```

Comment:
the key is `string`, why a reference equality instead of ordinal?

Reply:
i was keepign the same logic as GetTypeByMetadataName.  It appears as if ref-equality works as well, presumably because the strings loaded for thse attributes are only loaded once and then can be checked with ref-equality from that point on.

Reply:
Well now I'm curiosu why GetTypeByMetadataName uses that type of comparer ...

Reply:
my guess is that in practice thse are all constants that people pass in (or get from System.Type.Fullname) so it speeds thigns up slightly by just assuming you get the same instances.  Faster hashing/lookup, at the potential (but unlikely) cost of less caching if you actually had two different instances.